ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -312,6 +312,45 @@ public QueryMetadata getQueryExecutionPlan(final Query query, final KsqlConfig k
     }
   }
 
+  public List<Pair<String, Statement>> parseStatements(
+      final String queriesString,
+      final MetaStore tempMetaStore
+  ) {
+    try {
+      final MetaStore tempMetaStoreForParser = tempMetaStore.clone();
+      final KsqlParser ksqlParser = new KsqlParser();
+
+      final List<SqlBaseParser.SingleStatementContext> parsedStatements
+          = ksqlParser.getStatements(queriesString);
+      final List<Pair<String, Statement>> queryList = new ArrayList<>();
+
+      for (SqlBaseParser.SingleStatementContext singleStatementContext : parsedStatements) {
+        final Pair<Statement, DataSourceExtractor> statementInfo = ksqlParser.prepareStatement(
+            singleStatementContext,
+            tempMetaStoreForParser
+        );
+        final String statementString = getStatementString(singleStatementContext);
+        Statement statement = statementInfo.getLeft();
+        if (StatementRewriteForStruct.requiresRewrite(statement)) {
+          statement = new StatementRewriteForStruct(
+              statement,
+              statementInfo.getRight())
+              .rewriteForStruct();
+        }
+        buildSingleQueryAst(
+            statement,
+            getStatementString(singleStatementContext),
+            tempMetaStore,
+            tempMetaStoreForParser
+        );
+        queryList.add(new Pair<>(statementString, statement));
+      }
+      return queryList;
+    } catch (Exception e) {
+      throw new ParseFailedException("Exception while processing statements :" + e.getMessage(), e);
```

Comment:
Should we include the statements for context?

Reply:
Added statement text for the error message.